test/req/steps_test.exs

```diff
@@ -805,6 +805,76 @@ defmodule Req.StepsTest do
     assert log =~ "will retry in 1ms, 2 attempts left"
   end
 
+  @tag :capture_log
+  test "retry: default log_level", c do
+    adapter = fn request ->
+      request = Req.Request.update_private(request, :attempt, 0, &(&1 + 1))
+      attempt = request.private.attempt
+
+      response =
+        case attempt do
+          0 ->
+            Req.Response.new(status: 500, body: "oops")
+
+          1 ->
+            Req.Response.new(status: 200, body: "ok")
+        end
+
+      {request, response}
+    end
+
+    request = Req.new(adapter: adapter, url: c.url, retry_delay: 1)
+    log = ExUnit.CaptureLog.capture_log(fn -> Req.get!(request) end)
+
+    assert log =~ "[error]"
```

Comment:
We had this failure:

```
code:  assert log =~ "[info] retry: got response with status 500, will retry in 1ms, 3 attempts lef"
     left:  "\n22:41:35.585 [info]  retry: got response with status 500, will retry in 1ms, 3 attempts left\n"
     right: "[info] retry: got response with status 500, will retry in 1ms, 3 attempts lef"
     stacktrace:
       test/req/steps_test.exs:853: (test)
```

In short, `[info] retry` vs `[info]  retry:`. No idea where the extra space came from. Do you know? I'd rather keep these tests a bit stricter given we run async: true and thus they may accidentally pass if some other test just logs the log tag.

Reply:
I have no idea where the extra space came from. The tests were passing on my machine.. Let me look into getting it working with a regex.